src/consensus/tx_verify.cpp

```diff
@@ -401,22 +415,27 @@ bool Consensus::CheckTxAssets(const CTransaction& tx, CValidationState& state, c
                                      "bad-txns" + strError);
                 }
             }
+        }
+    }
 
+    for (const auto& outValue : totalOutputs) {
+        if (!totalInputs.count(outValue.first)) {
+            std::string errorMsg;
+            errorMsg = strprintf("Bad Transaction - Trying to create outpoint for asset that you don't have: %s", outValue.first);
+            return state.DoS(100, false, REJECT_INVALID, "bad-tx-inputs-outputs-mismatch " + errorMsg);
```

Comment:
copy/pasted error code

Reply:
The return statements are the same yes, but they return different messages.